core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -105,11 +105,11 @@ public static JsonNode get(String property, JsonNode node) {
   }
 
   public static int getInt(String property, JsonNode node) {
-    Preconditions.checkArgument(node.has(property), "Cannot parse missing int %s", property);
+    Preconditions.checkArgument(node.has(property), "Cannot parse missing int: %s", property);
     JsonNode pNode = node.get(property);
     Preconditions.checkArgument(
-        pNode != null && !pNode.isNull() && pNode.isNumber(),
-        "Cannot parse %s to an integer value: %s",
+        pNode != null && !pNode.isNull() && pNode.isIntegralNumber() && pNode.canConvertToInt(),
+        "Cannot parse to an integer value: %s: %s",
```

Comment:
Nit: The usage of double `:` here is a bit odd to me.

Maybe `Cannot parse integer value for property: (%s, %s)`?

Reply:
that's why I proposed using single quotes in https://github.com/apache/iceberg/pull/5434 because readability suffers with double `:`

Reply:
I think the double `:` is good.

Reply:
I’m ok with it. I don’t have a better alternative so I’m happy with that.

Reply:
Also seeing it as `propertyName: value` written out makes the double colon make a lot more sense. Thanks @nastra.